### PR TITLE
[luci/service]Add UnidirectionalSequenceLSTM on luci/service

### PR DESCRIPTION
**Pull request by struss**
Parent Issue : #4151
Draft PR : #4260

This commit add Type Inference/Shape Inference on luci.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 1]
```diff
@@ -2377,10 +2393,15 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
...
 
-  loco::NodeShape visit(const luci::CircleUnpack *node) final { return infer_unpack(node); }
+  loco::NodeShape visit(const luci::CircleUnidirectionalSequenceLSTM *node) final
+  {
+    return infer_unidirectionalsequencelstm(node);
+  }
 
   loco::NodeShape visit(const luci::CircleUnique *node) final { return infer_unique(node); }
 
+  loco::NodeShape visit(const luci::CircleUnpack *node) final { return infer_unpack(node); }
```
Comment (seanshpark) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:2403]: I think this change be in separate PR :)

Comment (struss) [reply compiler/luci/service/src/CircleShapeInferenceRule.cpp:2403]: Okay. When I add a this, I found this is not alphabetic order. so I just thought revise this, but this is not a part of adding new OP. so, I will rollback it.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 2]
```diff
@@ -1608,6 +1608,22 @@ loco::NodeShape infer_unpack(const luci::CircleUnpack *node)
   return loco::NodeShape{output_shape};
 }
 
+loco::NodeShape infer_unidirectionalsequencelstm(const luci::CircleUnidirectionalSequenceLSTM *node)
+{
+  auto input_shape = loco::shape_get(node->input()).as<loco::TensorShape>();
+  auto recurrent_to_output_weights =
+      loco::shape_get(node->recurrent_to_output_weights()).as<loco::TensorShape>();
+  auto rank = static_cast<int32_t>(input_shape.rank());
```
Comment (mhs4670go) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:1616]: Is there any reason of this `static_cast`?

Comment (struss) [reply compiler/luci/service/src/CircleShapeInferenceRule.cpp:1616]: `input_shape.rank()` is a type of `uint32_t`. so, I just use static_cast.

Comment (mhs4670go) [reply compiler/luci/service/src/CircleShapeInferenceRule.cpp:1616]: @struss got it. But it might as well goes on as it is without casting because rank shouldn't be minus value, which is why most of `rank` methods returns `uint32_t` I think.

Comment (struss) [reply compiler/luci/service/src/CircleShapeInferenceRule.cpp:1616]: Oh, I will update without casting. thanks for pointing out!

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 3]
```diff
@@ -1608,6 +1608,22 @@ loco::NodeShape infer_unpack(const luci::CircleUnpack *node)
...
 
+loco::NodeShape infer_unidirectionalsequencelstm(const luci::CircleUnidirectionalSequenceLSTM *node)
+{
+  auto input_shape = loco::shape_get(node->input()).as<loco::TensorShape>();
+  auto recurrent_to_output_weights =
+      loco::shape_get(node->recurrent_to_output_weights()).as<loco::TensorShape>();
+  auto rank = input_shape.rank();
+  loco::TensorShape output_shape;
+  output_shape.rank(rank);
+  for (int32_t i = 0; i < rank - 1; i++)
```
Comment (seanshpark) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:1619]: `rank` is `uint32_t` then this may drop compilation error of type mismatch.

Comment (struss) [reply compiler/luci/service/src/CircleShapeInferenceRule.cpp:1619]: Oh I will fix `i` as `uint32_t`.

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by mhs4670go — APPROVED** [review]
LGTM
